experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -15,51 +15,160 @@
  */
 
 import { AggregationTemporality } from './AggregationTemporality';
-import { MetricExporter } from './MetricExporter';
 import { MetricProducer } from './MetricProducer';
+import { MetricData } from './MetricData';
+
+export type ReaderOptions = {
+  timeoutMillis?: number
+}
+
+export type ReaderCollectionOptions = ReaderOptions;
+
+export type ReaderShutdownOptions = ReaderOptions;
+
+export type ReaderForceFlushOptions = ReaderOptions;
+
+/**
+ * Error that is thrown on timeouts (i.e. timeout on forceFlush or shutdown)
+ */
+export class ReaderTimeoutError extends Error {
+  constructor(message?: string) {
+    super(message);
+    Object.setPrototypeOf(this, ReaderTimeoutError.prototype);
```

Comment:
Yes. This is a workaround for [this](https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work). It can be a problem when using ES5 as a target, as `(new ReaderTimeoutError("message")) instanceof ReaderTimeoutError` will always evaluate to `false`.

Reply:
A comment with the link to the explanation will be very helpful in the case.

Reply:
Good idea, thanks! Added in 22e886a.